### PR TITLE
 Add library-level enforcement of component field limits

### PR DESCRIPTION
**Pull request by krittick**
## Summary

**WIP, still needs robust testing**

This adds library-level enforcement of limits on component fields, such as min/max values, label/placeholder lengths, and others. This helps reduce the chances of sending invalid data to the API.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by krittick** [comment]
Working on finishing it up today.

**Comment by krittick** [comment]
This is ready for review. Pretty sure I included all of the component limits, but let me know if I missed any.

**Review by Lulalaby — DISMISSED** [review]

File: discord/ui/input_text.py [hunk 1]
```diff
@@ -138,7 +152,9 @@ def min_length(self) -> Optional[int]:
     @min_length.setter
     def min_length(self, value: Optional[int]):
         if value and not isinstance(value, int):
-            raise TypeError(f"min_length must be None or int not {value.__class__}")  # type: ignore
+            raise TypeError(f"min_length must be None or int not {value.__class__.__name__}")  # type: ignore
+        if 0 <= value <= 4000:
+            raise ValueError("min_length must be between 0 and 4000")
```
Comment (Dorukyum) [inline discord/ui/input_text.py:157]: Wouldn't `value` already be between 0 and 4000 in this case?

Comment (krittick) [reply discord/ui/input_text.py:157]: Awkwaaaaard. Updated to the correct logic there and in `max_length.setter`

**Review by Lulalaby — APPROVED** [review]

File: discord/ui/input_text.py [hunk 2]
```diff
@@ -63,9 +63,19 @@ def __init__(
         row: Optional[int] = None,
     ):
         super().__init__()
-        custom_id = os.urandom(16).hex() if custom_id is MISSING else custom_id
-        if not (isinstance(custom_id, str) or custom_id is None):
+        if len(label) > 45:
+            raise ValueError("label must be 45 characters or fewer")
+        if 0 <= min_length <= 4000:
```
Comment (Dorukyum) [inline discord/ui/input_text.py:68]:
```suggestion
        if min_length < 0 or min_length > 4000:
```

File: discord/ui/input_text.py [hunk 3]
```diff
@@ -63,9 +63,19 @@ def __init__(
...
+            raise ValueError("min_length must be between 0 and 4000")
+        if 0 < max_length <= 4000:
```
Comment (Dorukyum) [inline discord/ui/input_text.py:70]:
```suggestion
        if max_length < 0 or max_length > 4000:
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Dorukyum — APPROVED** [review]

**Comment by Dorukyum** [comment]
Looks good

**Comment by krittick** [comment]
> Looks good

I do know math, I promise. Usually.

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**